src/ray/raylet/worker_pool.h

```diff
@@ -39,6 +40,56 @@ namespace raylet {
 using WorkerCommandMap =
     std::unordered_map<Language, std::vector<std::string>, std::hash<int>>;
 
+/// \struct WorkerCacheKey
+///
+/// Struct used to cache workers, keyed by runtime_env. The dynamic_options and
+/// override_environment_variables will be unified into runtime_env in the future.
+struct WorkerCacheKey {
+  std::vector<std::string> dynamic_options;
+  std::unordered_map<std::string, std::string> override_environment_variables;
+  std::string serialized_runtime_env;
+  mutable std::size_t hash_ = 0;
+
+  bool operator==(const WorkerCacheKey &k) const {
+    return dynamic_options == k.dynamic_options &&
```

Comment:
Why don't you just compare hash?

Reply:
Good point.  Earlier when I was working on this PR WorkerCacheKey was actually going to be used as a key to an STL hash map, which requires operator== to be implemented in this way to deal with hash collisions.  But since WorkerCacheKey isn't being used like this anymore, we should just compare the hashes